code/modules/mapping/mapping_helpers.dm

```diff
@@ -1359,3 +1359,35 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 
 	engraved_wall.AddComponent(/datum/component/engraved, engraving["story"], FALSE, engraving["story_value"])
 	qdel(src)
+
+/obj/effect/mapping_helpers/emergency_shuttle_event_override
+	name = "emergency shuttle events override"
+	icon_state = "shuttleeventoverride"
+	late = TRUE
+	/// List of event paths to run
+	var/list/events_override = list()
+
+/obj/effect/mapping_helpers/emergency_shuttle_event_override/LateInitialize()
+	var/obj/docking_port/mobile/port = SSshuttle.emergency
+	if(!port)
+		qdel(src)
+		return
+	port.event_list.Cut()
+	change_paths(port)
+	qdel(src)
+
+/obj/effect/mapping_helpers/emergency_shuttle_event_override/proc/change_paths(obj/docking_port/mobile/port)
```

Comment:
Check out our guy on [autodocccing](https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md)! Generally, above new procs you expect others to use and new objects, you put something explaining what it does and how it works
`/// Adds preset shuttle events to the emergency shuttle to run during the escape sequence`
`/obj/effect/.../proc/blabla(...)`